### ISSUE TITLE
Content improvements for API Docs overview page

### ISSUE THREAD

**Issue by thomaswilburn** [canjs/canjs#4439]
Some notes, work in progress:

### General

* Syntax is inconsistent--some places use old-school functions, some use arrow functions. Class decorators appear but aren't explained. 
* Where promises are used, would that code be clearer with async/await?
* On syntax, one of the things I really like from PouchDB's documentation is that they offer tabs for callback/Promise/async version on the code samples, so that people can see what it looks like in any environment.
* It's not always clear why some properties/methods are TitleCase and some are camelCase (`type` vs. `Type` in DefineMap seems like a particularly gruesome edge case). Would be good to have more clarity on that called out.
* "Run in your browser" opened about a million tabs with empty CodePens in Firefox, so I think there's some loops or duplicate event listeners in there somewhere.
* Is it policy not to refer to other frameworks? I think some of this might be made easier with comparisons to libraries that the reader might have used before, either in a larger section or when talking about specific concepts.

### Observables

This section should probably be called ViewModel instead, in keeping with the walkthrough flow. 

DefineMap is introduced here and used throughout the documentation, and it's very cool functionality, but it may be confusing for new people to the framework--and the name itself is opaque to understanding.

- Suggest revising the intro language:
   > The ViewModel of a component is extended from the DefineMap type, which creates an Observable object. DefineMaps provide a number of useful features, such as type validation and computed values, and they emit events when their properties are altered. Calling `DefineMap.extend()` returns a constructor for a custom subclass of DefineMap.
- Shorten the code example--it probably doesn't need to be a complete TodoJS implementation.
- Similar language for DefineList.
- The Ecosystem and Infrastructure API sections are a little unclear in relationship, and I don't know why they're collapsed to start.
  - Why would I use `observe()` or `observe.Object` if the Define types exist?
  - There's no demonstration of actually observing these objects, just setting and getting properties.
- Not technically the overview page, but when I click through to the docs for Definemap, the `@can.name` symbol convention is used but never explained--should we have a section for conventions somewhere? Did I maybe miss that? The "Reading the API Docs" guide wasn't very helpful to me.

### Views

- The view syntax demos are great, but could probably use a one-liner comment or header to set up context for each, instead of assuming that developers can intuit the point (they probably can, but it's more work than it needs to be).
- Do can-stache-bindings for properties rely on DefineMap or observables, or will they work with POJO? Would be nice to have a note about the underlying mechanism here.
- When we say that custom elements support data/event bindings, is that specific to CanJS implementations, or can we mix in native custom elements? Could I listen to the `amp-resize` event on a responsive iframe, for example? I may be the only person who cares about this now, but it could be an issue in the future.

**Comment by justinbmeyer** [comment]
>  Could I listen to the amp-resize event on a responsive iframe, for example?

Yes. That should work as long as it's dispatched as a custom event `elem.dispatchEvent(event);`

**Comment by justinbmeyer** [comment]
> Do can-stache-bindings for properties rely on DefineMap or observables, or will they work with POJO? Would be nice to have a note about the underlying mechanism here.

Yes, they rely on observables.  The observables need to support the `canReflect.onKeyValue` symbol:  https://canjs.com/doc/can-reflect/observe.onKeyValue.html

**Comment by justinbmeyer** [comment]
>  Not technically the overview page, but when I click through to the docs for Definemap, the @can.name symbol convention is used but never explained

Cam you explain more about this?

**Comment by justinbmeyer** [comment]
> It's not always clear why some properties/methods are TitleCase and some are camelCase (type vs. Type in DefineMap seems like a particularly gruesome edge case). Would be good to have more clarity on that called out.

When we expect a constructor function, we capitalize.  For example: `Type: Animal`.  The lower-case version is used to provide a function that returns the type.  For example: `type: function(x){ return x+2; }`.  `Type` and `Default` are shorthands:

```js
Type: Animal,
// same as
type: function(v){ return v instanceof Animal? v : new Animal(v) }
```

**Comment by justinbmeyer** [comment]
> Is it policy not to refer to other frameworks? I think some of this might be made easier with comparisons to libraries that the reader might have used before, either in a larger section or when talking about specific concepts.

Imo, it would be odd in the API docs, but there's a comparison page that we've long wanted to work on.

**Comment by thomaswilburn** [comment]
Re: symbol naming convention, if I look [here](https://canjs.com/doc/can-define/map/map.html#Mixed_ininstancemethodsandproperties), there's a number of mix-in props that use symbols as their name, and are referenced with an `@` at the start. I'd never seen that convention before, so I didn't know if it was referring to an instance variable from the superclass like Ruby, or something else. It took some clicking through to figure out that it seems to be the way symbols are referenced.

**Comment by justinbmeyer** [comment]
So MDN uses `@@` (Ex: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/@@iterator)... but I felt that was overkill and shortened to `@`.  I think there are some places still using `@@`.

Maybe we could have the hover show the type:

![image](https://user-images.githubusercontent.com/78602/44810156-79b2b880-ab96-11e8-8f33-510cc942f316.png)

Have you noticed that hover?

**Comment by thomaswilburn** [comment]
I didn't notice the hover. I worry that hiding things behind the `title` attribute is an [accessibility concern](https://developer.paciellogroup.com/blog/2012/01/html5-accessibility-chops-title-attribute-use-and-abuse/), and also makes it hard if anyone is looking these up on a mobile device (which does, admittedly, happen). On MDN, there's at least a reference to the Symbol type (or to `Symbol.iterator`) nearby in the plaintext.
